mixer/adapter/signalfx/signalfx.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signalfx
+
+//go:generate $GOPATH/src/istio.io/istio/bin/mixer_codegen.sh -f mixer/adapter/signalfx/config/config.proto
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/signalfx/golib/sfxclient"
+
+	"istio.io/istio/mixer/adapter/signalfx/config"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/metric"
+)
+
+type (
+	builder struct {
+		metricTypes map[string]*metric.Type
+		config      *config.Params
+	}
+
+	handler struct {
+		env                adapter.Env
+		ctx                context.Context
+		cancel             func()
+		scheduler          *sfxclient.Scheduler
+		registry           *registry
+		metricTypes        map[string]*metric.Type
+		ingestURL          string
+		accessToken        string
+		intervalSeconds    uint32
+		metricConfigByName map[string]*config.Params_MetricConfig
+		logger             adapter.Logger
+	}
+)
+
+// ensure types implement the requisite interfaces
+var _ metric.HandlerBuilder = &builder{}
+var _ metric.Handler = &handler{}
+
+///////////////// Configuration-time Methods ///////////////
+
+// adapter.HandlerBuilder#Build
+func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler, error) {
+	confsByName := make(map[string]*config.Params_MetricConfig, len(b.config.Metrics))
+	for i := range b.config.Metrics {
+		confsByName[b.config.Metrics[i].Name] = b.config.Metrics[i]
+	}
+
+	env.Logger().Infof("Building SignalFx adapter")
+
+	h := &handler{
+		env:                env,
+		metricTypes:        b.metricTypes,
+		ingestURL:          b.config.IngestUrl,
+		accessToken:        b.config.AccessToken,
+		intervalSeconds:    b.config.IntervalSeconds,
+		metricConfigByName: confsByName,
+		logger:             env.Logger(),
+	}
+
+	return h, h.Init()
+}
+
+// adapter.HandlerBuilder#SetAdapterConfig
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	b.config = cfg.(*config.Params)
+
+	if b.config.IntervalSeconds == 0 {
+		b.config.IntervalSeconds = 10
```

Comment:
Could you make this a constant at the top of the file?

Reply:
Actually, this case should be handled in the DefaultConfig value returned by GetInfo, no need to do this assignment here.